resources/benchmark-runner.mjs

```diff
@@ -36,15 +36,28 @@ class Page {
         });
     }
 
-    querySelector(selector) {
-        const element = this._frame.contentDocument.querySelector(selector);
+    getParent(path) {
+        if (!path)
+            return this._frame.contentDocument;
+
+        const parent = path.reduce((root, selector) => {
+            const node = root.querySelector(selector);
+            return node.shadowRoot ?? node;
+        }, this._frame.contentDocument);
+
+        return parent;
+    }
+
+    querySelector(selector, path) {
+        const element = this.getParent(path).querySelector(selector);
+
         if (element === null)
             return null;
         return this._wrapElement(element);
     }
 
-    querySelectorAll(selector) {
-        const elements = Array.from(this._frame.contentDocument.querySelectorAll(selector));
+    querySelectorAll(selector, path) {
```

Comment:
Maybe specify the default value of []?

Reply:
I'm checking in the `getParent` function: 

```
if (!path)
            return this._frame.contentDocument;
````

Reply:
Yeah, I know that but I think specifying [] here and not adding that special case would be better.

Reply:
ok - added defaults!